pkg/loop/telem.go

```diff
@@ -30,6 +56,61 @@ func SetupTelemetry(registerer prometheus.Registerer) GRPCOpts {
 	return GRPCOpts{DialOpts: dialOptions(registerer), NewServer: newServerFn(registerer)}
 }
 
+// SetupTracing initializes open telemetry with the provided config.
+// It sets the global trace provider and opens a connection to the configured collector. 
+func SetupTracing(config TracingConfig) {
+	log, err := logger.New()
+	
+	ctx := context.Background()
+	ctx, cancel := context.WithTimeout(ctx, time.Second)
+
+	// Enough to shutdown the underlying connection since DialContext is used in blocking mode
+	defer cancel()
+	conn, err := grpc.DialContext(ctx, config.CollectorTarget,
+		// Note the use of insecure transport here. TLS is recommended in production.
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		// safe usage in this case https://github.com/grpc/grpc-go/blob/master/Documentation/anti-patterns.md#dialing-in-grpc
+		grpc.WithBlock(),
+	)
+	if err != nil {
+		log.Errorf("Connecting to OTEL collector failed: %v", err)
+	}
+
+	// Set up a trace exporter
+	// Shutting down the traceExporter will not shutdown the underlying connection.
+	traceExporter, err := otlptracegrpc.New(ctx, otlptracegrpc.WithGRPCConn(conn))
+	if err != nil {
+		log.Errorf("failed to create trace exporter: %w", err)
+		return
+	}
+
+	attributes := []attribute.KeyValue{
+		semconv.ServiceNameKey.String("cl-node"),
+		semconv.ProcessPIDKey.Int(os.Getpid()),
+		// TODO: figure out how to wire node version through?
+		semconv.ServiceVersionKey.String("0.0.1"),
```

Comment:
Don't we want the LOOPP program version here, rather than the core node?
The runtime should have build info that we can tap in to: https://pkg.go.dev/runtime/debug#ReadBuildInfo

Reply:
I believe we would want either the LOOP program version or the code node version. `ReadBuildInfo` will work great - we can get the module version of the binary through the returned object of type BuildInfo.

Reply:
We might need the full package name for the `ServiceNameKey`, to be able to distinguish between main packages within the same module. (I'm assuming the short program names won't be consistent or reliable?)